pkg/runtime/adoption_reconciler.go

```diff
@@ -466,3 +478,26 @@ func NewAdoptionReconciler(
 		},
 	}
 }
+
+// LogAdoptedResource returns a logger with log values set for the resource's kind,
+// namespace, name, etc
+func LogAdoptedResource(
+	log logr.Logger,
+	res *ackv1alpha1.AdoptedResource,
+	additionalValues ...interface{},
+) logr.Logger {
+	ns := res.Namespace
+	resName := res.Name
+	generation := res.Generation
+	kind := res.Kind
```

Comment:
This is always going to be AdoptedResource. I think a more useful included value might be `res.Kubernetes.Kind` and `res.Kubernetes.Group`, which represent the target CR that the adopted resource reconciler will end up creating.

https://github.com/aws-controllers-k8s/runtime/blob/10d1b5339b6030fcaedf65d84e637d12a8779ab1/apis/core/v1alpha1/identifiers.go#L29-L32

Reply:
Can we add this in future?

this will add value when we have more detailed error messages